### ISSUE TITLE
no make_unexpected with nsel_USES_STD_EXPECTED

### ISSUE THREAD

**Issue by psyinf** [nonstd-lite/expected-lite#66]
I'm using the library indirectly via [BehaviorTree.CPP](https://github.com/BehaviorTree/BehaviorTree.CPP).
Using a recent clang(17.0.3)/gcc(14.1) the code doesn't compile anymore due to make_unexpected not being available. 

I currently patched this locally by adding this into the `#if nsel_USES_STD_EXPECTED` path (~ line 253):

```template <typename E>
constexpr auto make_unexpected(E&& value)
{
  return std::unexpected(std::forward<E>(value));
}
```

If wanted, I can submit a PR with the change or maybe I'm doing something wrong here?

**Comment by martinmoene** [comment]
Thanks for the heads-up.

Function `make_expected()` was part of the proposal for `std::expected` up to and including [revision 3](http://wg21.link/p0323r3) and is available with the _`nonstd::expected` implementation_ provided `nsel_P0323R <= 3` (see section [Configuration](https://github.com/martinmoene/expected-lite?tab=readme-ov-file#configuration)).

Consequently `make_expected()` is not available when `std::expected` is used for `nonstd::expected` when available and requested as you experience with the recent compilers. Note: in this case it will not help to `-Dnsel_P0323R=3`

The idea of _excpected-lite_ being a migration path to `std::expected`, suggests (indeed) that `make_expected()` and comparison functions should be available in the case of `std::expected` being used for `nonstd::expected` AND `-Dnsel_P0323R=3` (2 to also include comparison functions).

I'll look into that, unsure how quickly I'll get at it.

**Comment by psyinf** [comment]
Thanks for looking into it. 

The challenge here lies in the _expected-lite_ being used by other libraries taking the  liberty in using the P0323R suggested feature throughout their codebase. My suggestion is merely a quick-fix, trying to forward the ´std::unexpected{}´-idiom to the ´make_unexpected´ functionality.

**Comment by martinmoene** [comment]
I'll add `make_expected()` ~~un~~conditionally (assuming `nsel_P0323R` is already used) for use with `std::expected()`.

Have yet to correct the test suite for use with `std::expected`, which wasn't yet available at the time of writing.

**Comment by martinmoene** [comment]
Argh, I now realise it's (also) about `make_unexpected()`...

Will continue tomorrow.

**Comment by psyinf** [comment]
@martinmoene  Thanks for tackeling the issue. I'll try to the current head in my scenario and advise the authors of BehaviorTree.CPP to use a new release, as soon as available.

**Comment by martinmoene** [comment]
Currently adapting the test suite for when `std:: exception` is  being used.

Will make a release after that, presumably v0.8.0.

**Comment by psyinf** [comment]
Just tested compliation with gcc and clang locally, looks good, so the issue can be closed, 
Thank you again for the speedy solution

**Comment by martinmoene** [comment]
Thanks for your feedback & you're welcome.

**Comment by martinmoene** [comment]
@psyinf Released _expected-lite_ [v0.8.0](https://github.com/martinmoene/expected-lite/releases/tag/v0.8.0).
